pkg/http/download.go

```diff
@@ -0,0 +1,43 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"go.uber.org/zap"
+)
+
+// DownloadFile downloads a file from the specified URL and stores it to the given path.
+func DownloadFile(ctx context.Context, url, path string) error {
+	zap.S().Infof("Downloading file from '%s' to '%s'...", url, path)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return fmt.Errorf("creating request: %w", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
No surprise, I've never used this library in Go. Do we want to explicitly add some sort of timeout to this call (assuming it's even supported)? I worry about the possibility of EIB "hanging" because it's trying to download and for whatever reason, the connection is taking forever.

Reply:
This is what the `ctx context.Context` field is included. The caller will determine whether or how it'd want to manage the download.

You can refer to the [package documentation](https://pkg.go.dev/context) or this [blog post](https://go.dev/blog/context) for further explanation and examples.

Reply:
Short example of how this will work for us:

```
ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
defer cancel()
			
err := DownloadFile(ctx, <url>, <path>)
// Error handling
```

If the `http.Do(req)` call (in `DownloadFile`) takes longer than 10 seconds, it will automatically fail with a `context deadline exceeded` error.

Reply:
Ahhh, ok, so there is a hook for it. Thanks for the clarification.